contracts/marketplace/src/execute.rs

```diff
@@ -691,6 +703,40 @@ pub fn execute_accept_bid(
     Ok(res.add_event(event))
 }
 
+pub fn execute_reject_bid(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    collection: Addr,
+    token_id: TokenId,
+    bidder: Addr,
+) -> Result<Response, ContractError> {
+    nonpayable(&info)?;
+    only_owner(deps.as_ref(), &info, &collection, token_id)?;
+
+    let bid_key = bid_key(&collection, token_id, &bidder);
+
+    let bid = bids().load(deps.storage, bid_key.clone())?;
+    if bid.is_expired(&env.block) {
+        return Err(ContractError::BidExpired {});
+    }
+
+    bids().remove(deps.storage, bid_key)?;
+
+    let refund_msg = BankMsg::Send {
+        to_address: bidder.to_string(),
+        amount: vec![coin(bid.price.u128(), NATIVE_DENOM.to_string())],
+    };
+
+    let event = Event::new("reject-bid")
+        .add_attribute("collection", collection.to_string())
+        .add_attribute("token_id", token_id.to_string())
+        .add_attribute("bidder", bidder)
+        .add_attribute("price", bid.price.to_string());
```

Comment:
This would probably require an indexer change. If we name it the same `remove-bid` won't require any change but we could add a `rejected-> "true"` attribute.

It would be a bit weird tho.

Reply:
Adding support for `reject-bid` is easy, I don't mind. I just need to know when this is deployed on testnet with a block height with such an event so I add a test.

Reply:
@jhernandezb I'd rather not write contracts to appease the indexer. The indexer should adapt to handle contract changes based on version. @penso Thanks!